tests/test_filesystem.py

```diff
@@ -17,67 +18,120 @@ class TestFileSystem(unittest.TestCase):
     """
 
     def setUp(self):
-        self.cfg_file = tempfile.NamedTemporaryFile(mode="w", suffix=".cfg", delete=False)  # pylint: disable=R1732
         self.test_username = "test_username"
+        self.callback_url = "callback_url"
+        self.jupyterhub_user_data_dir = "/jupyterhub_user_data"
 
     def tearDown(self):
         utils.clear_handlers_instances()
-        os.unlink(self.cfg_file.name)
 
-    def test_manage_user_workspace(self):
+    @patch("cowbird.api.webhooks.views.requests.head")
+    def test_manage_user_workspace(self, mock_head_request):
         """
         Tests creating and deleting a user workspace.
         """
-        with tempfile.TemporaryDirectory() as workspace_dir:
-            with self.cfg_file as f:
-                f.write(yaml.safe_dump({"handlers":
-                                        {"FileSystem": {"active": True, "workspace_dir": workspace_dir}}}))
-            app = utils.get_test_app(settings={"cowbird.config_path": self.cfg_file.name})
+        with tempfile.TemporaryDirectory() as workspace_dir, \
+                tempfile.NamedTemporaryFile(mode="w", suffix=".cfg") as cfg_file:
+            user_workspace_dir = Path(workspace_dir) / self.test_username
+            user_symlink = user_workspace_dir / NOTEBOOKS_DIR_NAME
+
+            cfg_file.write(yaml.safe_dump({
+                "handlers": {
+                    "FileSystem": {
+                        "active": True,
+                        "workspace_dir": workspace_dir,
+                        "jupyterhub_user_data_dir": self.jupyterhub_user_data_dir}}}))
+            cfg_file.flush()
+            app = utils.get_test_app(settings={"cowbird.config_path": cfg_file.name})
             data = {
                 "event": "created",
                 "user_name": self.test_username,
-                "callback_url": "callback_url"
+                "callback_url": self.callback_url
             }
             resp = utils.test_request(app, "POST", "/webhooks/users", json=data)
             utils.check_response_basic_info(resp, 200, expected_method="POST")
-            assert (Path(workspace_dir) / self.test_username).exists()
+            assert user_workspace_dir.exists()
+            assert os.path.islink(user_symlink)
+            assert os.readlink(user_symlink) == os.path.join(self.jupyterhub_user_data_dir, self.test_username)
+            assert utils.check_file_permissions(str(user_workspace_dir), "755")
```

Comment:
Will have to see if a `chown`/`check_file_owner` will be needed here following our discussion.
Same for all following checks.

See also [DAC-547](https://crim-ca.atlassian.net/jira/software/c/projects/DAC/issues/DAC-547).